### PR TITLE
#117 add vectors

### PR DESCRIPTION
**Pull request by AlexKay28**
transfer from multi armed bandits to the vector space representation of users and questions

### CODE REVIEW

**Comment by AlexKay28** [comment]
During the reaseach I found out a lot of different implementation difficulties. Also during the development I decided to put the research details into this repository -> https://github.com/AlexKay28/studentRL

When I will have finished and will know which model to use , I will continue integrating DRL into the project

File: backend/dbase/phrases.py [hunk 1]
```diff
@@ -11,7 +11,15 @@ class Phrase(db.Model):
     russian = db.Column(db.String(256), nullable=False)
     ukrainian = db.Column(db.String(256), nullable=False)
 
-    def __init__(self, id, level, english, french, russian, ukrainian):
+    def __init__(
```
Comment (alantishin) [inline backend/dbase/phrases.py:14]: describing type annotation for each argument 
will be helpful )

File: backend/utils/db_controller.py [hunk 2]
```diff
@@ -9,10 +9,16 @@
...
 from dbase.database_connector import DatabaseConnector
-from dbase.actions import Action
-from dbase.transitions import TransitionSuccess, TransitionShift
+from dbase.phrases import Phrase, PhraseVector
 
 from utils.helpers import get_phrase_links
+from utils.facade_api import FacadeAPI
+
+# set other services connection
+NLP_SERVICE_URL = os.environ.get("NLP_SERVICE_URL")
```
Comment (alantishin) [inline backend/utils/db_controller.py:18]: you can also have one config file 
https://softwareengineering.stackexchange.com/questions/360850/where-should-i-place-configuration-classes-in-onion-architecture

File: backend/utils/session_object.py [hunk 3]
```diff
@@ -196,20 +215,35 @@ def select_question(
...
 
-        if level > 0:
-            phrases_id = (
-                db.session.query(Phrase.id).filter(Phrase.level == level).distinct()
+        previous_phrases_vecs_query = sql_text_query(
+            f"""
+            with phrases_ids as (
+                SELECT phrase_id
+                FROM actions
+                WHERE uuid = '{uuid}'
```
Comment (alantishin) [inline backend/utils/session_object.py:223]: it is not safely to pass value into sql query this way
you can get sql injection

Comment (AlexKay28) [reply backend/utils/session_object.py:223]: oh gosh.. you're right!
I'll fix it! thanks

File: backend/dbase/phrases.py (hunk 1 above)
Comment (AlexKay28) [reply backend/dbase/phrases.py:14]: kk)

File: backend/utils/db_controller.py (hunk 2 above)
Comment (AlexKay28) [reply backend/utils/db_controller.py:18]: not really understand ( 
how can the link above help me?

File: backend/utils/session_object.py (hunk 3 above)
Comment (AlexKay28) [reply backend/utils/session_object.py:224]: I have added check if it's correct uuid. Dont have any fast and good idea about how to fix it in other way

**Pull request merged**